src/test/java/com/google/cloud/flink/bigquery/integration/FlinkReadFromQueryIntegrationTest.java

```diff
@@ -15,134 +15,118 @@
+    final Table sourceTable = flinkTableEnv.from(flinkSrcTable);
+    TableResult datatable = sourceTable.execute();
+    assertNotNull(datatable);
+    assertEquals(2, datatable.getTableSchema().getFieldCount());
   }
 
   @Test
   public void testBadSql() {
+    String flinkSrcTable = "FlinkSrcTable";
+    String sql =
+        "SELECT tagging, COUNT(*) countVal FROM ( SELECT SPLIT(tags, \"|\") tagging FROM `q-gcp-6750-pso-gs-flink-22-01.testDataset.posts_questions` a WHERE EXTRACT(YEAR FROM creation_date)>=2014 ), UNNEST(tags) tag GROUP BY 1 ORDER BY 2 DESC LIMIT 10 ";
     assertThrows(
         RuntimeException.class,
         () -> {
-          String bigqueryReadTable =
-              "q-gcp-6750-pso-gs-flink-22-01.wordcount_dataset.wordcount_output";
-          String flinkSrcTable = "FlinkSrcTable";
-          String flinkSrcTable1 = "FlinkSrcTable";
           String srcQueryString =
-              "CREATE TABLE " + flinkSrcTable1 + " (word STRING , word_count BIGINT)";
+              "CREATE TABLE " + flinkSrcTable + " (tag STRING,tag_count BIGINT)";
           flinkTableEnv.executeSql(
               srcQueryString
                   + "\n"
                   + "WITH (\n"
                   + "  'connector' = 'bigquery',\n"
                   + "  'format' = 'arrow',\n"
-                  + "  'table' = '"
-                  + bigqueryReadTable
+                  + "  'query' = '"
+                  + sql
                   + "',\n"
-                  + "  'filter' = 'word_count > 500',\n"
+                  + "  'maxParallelism' = '10',\n"
+                  + "  'materializationProject' = 'q-gcp-6750-pso-gs-flink-22-01',\n"
+                  + "  'materializationDataset' = 'testDataset',\n"
                   + "  'credentialsFile' = '"
                   + System.getenv("GOOGLE_APPLICATION_CREDENTIALS")
-                  + "' ,\n"
-                  + "  'selectedFields' = 'word_test' \n"
+                  + "'\n"
                   + ")");
-          Table result = flinkTableEnv.from(flinkSrcTable1);
-          Table datatable =
-              result
-                  .where($("word_count").isGreaterOrEqual(100))
-                  .select($("word"), $("word_count"));
-          TableResult tableapi = datatable.execute();
+          Table result = flinkTableEnv.from(flinkSrcTable);
+          result.execute();
           throw new FlinkBigQueryException("Column not found");
         });
   }
 
-  // We are passing all the configuration values and setting filter in flink and tableAPI both
-  // together.
   @Test
-  public void testReadFromQueryInternal() throws Exception {
-    String bigqueryReadTable = "q-gcp-6750-pso-gs-flink-22-01.wordcount_dataset.wordcount_output";
+  public void testReadFromQueryWithNewLine() {
+    String sql =
+        "SELECT tag, COUNT(*) countVal \n"
+            + "FROM ( SELECT SPLIT(tags, \"|\") tags FROM `q-gcp-6750-pso-gs-flink-22-01.testDataset.posts_questions` a \n"
```

Comment:
please test using bigquery public data only

Reply:
@davidrabinowitz , Getting permission issues while using the public data in query style

Reply:
Error message:
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 403 Forbidden
GET https://www.googleapis.com/bigquery/v2/projects/q-gcp-6750-pso-gs-flink-22-01/queries/20de8f5d-c6e6-461b-a502-b46bbab94fe3?location=us-central1&maxResults=0&prettyPrint=false
{
  "code" : 403,
  "errors" : [ {
    "domain" : "global",
    "message" : "Access Denied: Table bigquery-public-data:stackoverflow.posts_questions: User does not have permission to query table bigquery-public-data:stackoverflow.posts_questions.",
    "reason" : "accessDenied"
  } ],
  "message" : "Access Denied: Table bigquery-public-data:stackoverflow.posts_questions: User does not have permission to query table bigquery-public-data:stackoverflow.posts_questions.",
  "status" : "PERMISSION_DENIED"
}